docs/deployment_patterns/how_to_run_a_checkpoint_in_airflow.md

```diff
@@ -14,59 +14,254 @@ This guide will help you run a Great Expectations checkpoint in Apache Airflow,
 
-You can use a simple `BashOperator` in Airflow to trigger the checkpoint run. The following snippet shows an Airflow task for an Airflow DAG named `dag` that triggers the run of a checkpoint we named `my_checkpoint`:
+It’s recommended to specify a version in the `requirements.txt` file. To make use of the latest Great Expectations V3 API, you’ll need to specify a version `>=0.1.0`, and if you’re using Astronomer, you’ll need version `>=0.1.1`. In either case, the Great Expectations V3 API provider requires Airflow 2, so if you’re still running Airflow 1.X, upgrade that first.
+
+## Using the Operator
+
+The second step to use the operator is to import it into the DAG. The DAG file is a python file in the `dags/` folder of your Airflow project. To import the Great Expectations provider, add the following line at the top of your DAG file:
+
+`from great_expectations_provider.operators.great_expectations import GreatExpectationsOperator`
+
+Using the operator in a DAG is as simple as defining an instance of the `GreatExpectationsOperator` class and assigning it to a variable, like so:
+
+```python
+ge_data_context_root_dir_with_checkpoint_name_pass = GreatExpectationsOperator(
+    task_id="ge_data_context_root_dir_with_checkpoint_name_pass",
+    data_context_root_dir=ge_root_dir,
+    checkpoint_name="taxi.pass.chk",
+)
+
+ge_data_context_config_with_checkpoint_config_pass = GreatExpectationsOperator(
+    task_id="ge_data_context_config_with_checkpoint_config_pass",
+    data_context_config=example_data_context_config,
+    checkpoint_config=example_checkpoint_config,
+)
+```
+
+The above code creates two Great Expectations operators assigned to two distinct variables. Once defined, the operators must be put into the DAG, like so:
+
+```python
+ge_data_context_root_dir_with_checkpoint_name_pass >> ge_data_context_config_with_checkpoint_config_pass
+```
+
+The above code defines the DAG as two operations, the first being `ge_data_context_root_dir_with_checkpoint_name_pass` and the second `ge_data_context_config_with_checkpoint_config_pass`, which will run sequentially in that order.
+
+The operator has several optional arguments, but it always requires either a `data_context_root_dir` or a `data_context_config` and either a `checkpoint_name` or `checkpoint_config`.
+
+The `data_context_root_dir` should point to the `great_expectations` project directory previously generated. When running an Astronomer deployment, it is recommended that this directory is under the `include/` directory, which is automatically generated when creating an Astronomer project. When using a `data_context_config`, the `DataContextConfig` object must also be imported into the project file using:
+`from great_expectations.data_context.types.base import DataContextConfig`
```

Comment:
Let's include all imports together, since that's the natural way someone would write their DAG. We can then break the setup for `DataContextConfig` into its own header, since it's a noticeably different part of the setup.

Reply:
Well, so the one thing about this is that these imports (for `DataContextConfig` and `CheckpointConfig`) ideally don't go in the DAG, but their own config files. This is currently how they're exemplified in the GE example DAG on their repository (and the DAGs I wrote). I changed this section around a bunch, so might be best to make a decision after I push more changes.